content_sync/tasks_test.py

```diff
@@ -110,6 +112,36 @@ def test_sync_website_content_not_exists(api_mock, log_mock):
     api_mock.get_sync_backend.assert_not_called()
 
 
+def test_update_mass_build_pipelines(api_mock):
+    """Verify that mass_build_sites pipelines are upserted under the proper conditions"""
+    website = WebsiteFactory.create()
+    fake_date = factory.Faker("date_time", tzinfo=pytz.utc)
+    website.draft_publish_date = None
+    website.publish_date = None
+    tasks.update_mass_build_pipelines(website, VERSION_DRAFT)
+    api_mock.get_mass_build_sites_pipeline.assert_any_call(
+        VERSION_DRAFT, offline=False, prefix="", starter=""
+    )
+    api_mock.get_mass_build_sites_pipeline.assert_any_call(
+        VERSION_DRAFT, offline=True, prefix="", starter=""
+    )
+    assert api_mock.get_mass_build_sites_pipeline.call_count == 2
+    website.draft_publish_date = fake_date
+    tasks.update_mass_build_pipelines(website, VERSION_DRAFT)
+    assert api_mock.get_mass_build_sites_pipeline.call_count == 2
+    tasks.update_mass_build_pipelines(website, VERSION_LIVE)
+    api_mock.get_mass_build_sites_pipeline.assert_any_call(
+        VERSION_LIVE, offline=False, prefix="", starter=""
+    )
+    api_mock.get_mass_build_sites_pipeline.assert_any_call(
+        VERSION_LIVE, offline=True, prefix="", starter=""
```

Comment:
Seems like this is what the prefix will always default to in the `MassBuildSitesPipeline` code unless some other value is passed, so could leave it out.  Should work fine either way though.

Reply:
Really I should update the defaults, because when I ran this without them being specified they show up as "null" in Concourse which is something that should be avoided.